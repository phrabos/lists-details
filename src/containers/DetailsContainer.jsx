import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

const DetailsContainer = () => (
  <h1>hello</h1>
);

DetailsContainer.prototypes = {

};

export default DetailsContainer;
