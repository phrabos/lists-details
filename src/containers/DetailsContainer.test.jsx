import React from 'react';
import { render, screen } from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import DetailsContainer from './DetailsContainer';

const server = setupServer(
  rest.get('https://ac-vill.herokuapp.com/villagers/5f5fb4bbbfd05c2aed82e460', (req, res, ctx) => {
    return res(ctx.json({
      id: '5f5fb4bbbfd05c2aed82e460',
      name: 'Admiral',
      image: 'https://vignette.wikia.nocookie.net/animalcrossing/images/e/ed/Admiral_NH.png/revision/latest?cb=20200802081138',
      quote: 'Only quitters give up.',
      personality: 'cranky',
      skill: 'Writing about pickles',
      species: 'bird',
    }));
  })
);

describe('DetailsContainer', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('renders a list of characters on the page', async () => {
    render(<DetailsContainer match={{
      params: {
        id: '5f5fb4bbbfd05c2aed82e460'
      }
    }  
    }/>);

    screen.getByText('Loading...');

    const characterDiv = await screen.findByTestId('character-div');
    expect(characterDiv).toMatchSnapshot();
  });
});
