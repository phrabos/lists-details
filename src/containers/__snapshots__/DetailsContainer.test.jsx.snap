// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsContainer renders a list of characters on the page 1`] = `
<div
  data-testid="character-div"
>
  <img
    alt="Admiral"
    src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/ed/Admiral_NH.png/revision/latest?cb=20200802081138"
  />
  <p>
    Admiral
  </p>
  <p>
    Only quitters give up.
  </p>
  <p>
    cranky
  </p>
  <p>
    bird
  </p>
  <p>
    Writing about pickles
  </p>
</div>
`;
